File: results/dnd-rest-benchmark/lifecycle-runs/20260730T150315Z_pi_kimi-k2p7-code_typescript-nextjs/app/lib/maintenance.ts
/**
 * Process-global maintenance switch controlled by an authenticated DM.
 *
 * This lives outside of the SQLite store because readiness is a runtime
 * property of the current server process, not campaign-local persisted state.
 */

let maintenance = false;

export function getMaintenanceMode(): boolean {
  return maintenance;
}

export function setMaintenanceMode(value: boolean): void {
  maintenance = value;
}

File: results/dnd-rest-benchmark/lifecycle-runs/20260730T075550Z_pi_kimi-k2p7-code_ruby-sinatra/lib/maintenance.rb
# frozen_string_literal: true

# Process-global maintenance switch for the readiness probe.
module Maintenance
  @mutex = Mutex.new
  @enabled = false

  def self.enabled?
    @mutex.synchronize { @enabled }
  end

  def self.enabled=(value)
    @mutex.synchronize { @enabled = value }
  end
end
